### DISCUSSION TITLE
Order book imbalance strategy params

### DISCUSSION THREAD

**Discussion by artiko88** [nkaz001/hftbacktest#29] [General]
Hey!

I've been tinkering with an example strategy based on order book imbalance.
In the code of the strategy the derived imbalance is multiplied by a c1 variable which is set to 150 in the example. 
I was wondering what's the logic behind this parameter and wether one should try to tune it?

**Comment by nkaz001** [comment]
You can think of the half spread to c1 ratio as a threshold. I found it heuristically, or you can search for optimized parameters using grid search, random search or more advanced search algorithm such as evolutionary search.

It's my old research, but it could be helpful. https://github.com/nkaz001/algotrading-example/blob/master/crypto-scratch/btcusdt-standardization-market-making-backtest.ipynb

**Comment by artiko88** [comment]
> You can think of the half spread to c1 ratio as a threshold. I found it heuristically, or you can search for optimized parameters using grid search, random search or more advanced search algorithm such as evolutionary search.

@nkaz001,

When using those algorithms, is it necessary to run through the whole backtest or is it possible to somehow conduct a search _during_ the backtest? For example, i use linear regression for price prediction on a window of data collected during the backtest. I am thinking about conducting a search for c1 or other adjusting parameters during the test as well,

**Reply by nkaz001** [reply]
I think that could be possible, but you need to verify that your method is yielding expected results through backtesting.
